pubsubs/base.py

```diff
@@ -0,0 +1,89 @@
+""" Abstract base class, controls the interface for pub/sub implementations."""
+import logging
+import functools
+from abc import abstractmethod
+
+from interface_meta import quirk_docs
+from interface_meta import InterfaceMeta
+
+from pubsubs.subscriber import Subscriber
+
+
+class MessageQueue(metaclass=InterfaceMeta):
+    """ Abstract Base Class."""
+
+    INTERFACE_EXPLICIT_OVERRIDES = True
+    INTERFACE_RAISE_ON_VIOLATION = True
+    BACKENDS = None
+
+    def __init__(self, name, registry, backend, listeners, **config):
+        self.name = name
+        self.registry = registry
+        #: The name of the implementation
+        self.backend = backend
+        self.listeners = listeners
+        self.config = config
+
+    def subscribe(self, *topics):
+        """ Return a subscriber instance associated to the concrete
+        implementation.
+        """
+        self._connect()
+        topics = list(topics)
+
+        # Initialise new subscriber from the name of concrete class
+        return Subscriber.for_backend(self.backend)(
+            self.subscriber_config, topics, self._serializer
+        ).connect()
+
+    @quirk_docs("_publish")
+    def publish(self, topic, message):
+        """ Publish `message` to the `topic`."""
+        self._connect()
```

Comment:
You connect every time you publish a message??

Reply:
You're right I only need to `_connect()` when I instantiate the class